src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
+JSON::
++
+[source,json,indent=0]
+----
+include::example$manual/webapp/log4j2-single.json[tag=routing]
+----
+
+YAML::
++
+[source,yaml,indent=0]
+----
+include::example$manual/webapp/log4j2-single.yaml[tag=routing]
+----
 
-[source,java]
+Properties::
++
+[source,properties,indent=0]
+----
+include::example$manual/webapp/log4j2-single.properties[tag=routing]
 ----
-import java.io.IOException;
-import javax.servlet.AsyncContext;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+====
+
+[#multiple-contexts]
+==== Multiple logger contexts
+
+Since Log4j Core uses
+link:../javadoc/log4j-core/org/apache/logging/log4j/core/selector/ClassLoaderContextSelector.html[`ClassLoaderContextSelector`]
+by default, no configuration is needed to achieve multiple logger contexts in your application server:
+the classes of each classloader will use the logger context associated to the classloader.
+
+[TIP]
+====
+To provide a different configuration file for each logger context, you can add files named `log4j2<contextName>.xml` to the classpath of your application server.
+
+See <<log4jContextName>> and <<log4jConfiguration>> for more details.
+====
+
+Associating logger contexts to classloaders has however some limitations: **shared** libraries will not be able to use the per-application logger contexts.
+To overcome this limitation Log4j Core provides an alternative algorithm to determine the right logger context to choose: JNDI lookups.
+
+[#jndi-configuration]
+==== JNDI context selector
+
+Application servers set up the correct JNDI context as soon as they determine, which application will handle a request.
+Log4j Core allows the usage of JNDI to coordinate the usage of logger contexts in a Jakarta EE application server.
+In order to use this feature, you need to:
+
+. Set the
+xref:manual/systemproperties.adoc#log4j2.contextSelector[`log4j2.contextSelector`]
+Log4j configuration property to ``org.apache.logging.log4j.core.selector.JndiContextSelector``,
+. For security reasons you need to enable the selector, by setting the
+xref:manual/systemproperties.adoc#log4j2.enableJndiContextSelector[`log4j2.enableJndiContextSelector`]
+Log4j configuration property to `true`,
+. Each web application needs to configure the servlet context parameter `isLog4jContextSelectorNamed` to `true` and provide a value for the `log4jContextName` servlet context parameter and `java:comp/env/log4j/context-name` JNDI environment entry:
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/jndi.xml[tag=jndi]
```

Comment:
Mind providing a link to the actual source in the caption of the source block, please?

Reply:
Fixed in 0338eada282faaa12a16cb27b9e0e6c2198e90e2.